### PR TITLE
README.md: add reference to FAQ

### PR DESCRIPTION
**Pull request by maribu**
### Contribution description

At the bottom the related projects section in the README.me, this adds a reference to the in-detail comparison between RIOT and related OSes in our FAQ.

### Testing procedure

Proofreading and possible checking the output of the rendered version (easiest [in the branch](https://github.com/maribu/RIOT/tree/README.md/add-ref-to-detailed-comparison) corresponding to this PR).

### Issues/PRs references

Follow up to https://github.com/RIOT-OS/RIOT/pull/21052

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -91,6 +91,12 @@ in Rust following RIOT's goals and [vision](https://doc.riot-os.org/vision.html)
 - [RT-Thread](https://github.com/RT-Thread/rt-thread) is a microcontroller operating system with strong roots in China and a large community there, it has extensive support for MCUs from Chinese vendors, but also for western ones.
 - [Zephyr](https://github.com/zephyrproject-rtos/zephyr) is a general-purpose operating system for microcontrollers, shepherded by the Linux foundation.
 
+For more details on how RIOT compares to other projects, refer to the
+[comparison][faq-riot-vs-others] of RIOT with other operating systems for
+embedded and IoT devices [in our FAQ][faq-riot-vs-others].
+
+[faq-riot-vs-others]: https://www.riot-os.org/#faq-answer-12
```
Comment (kfessel) [inline README.md:98]: somehow the riot-os.org "lazy-loader" / "don't make me click" thingy is not able to deal with this (at least for me might be browser dependant)

Comment (maribu) [reply README.md]: Not ideal the latest fixup. But probably better than a non-working link.

File: README.md [hunk 2]
```diff
@@ -92,10 +92,10 @@ in Rust following RIOT's goals and [vision](https://doc.riot-os.org/vision.html)
 - [Zephyr](https://github.com/zephyrproject-rtos/zephyr) is a general-purpose operating system for microcontrollers, shepherded by the Linux foundation.
 
 For more details on how RIOT compares to other projects, refer to the
-[comparison][faq-riot-vs-others] of RIOT with other operating systems for
-embedded and IoT devices [in our FAQ][faq-riot-vs-others].
+comparison of RIOT with other operating systems for embedded and IoT devices
+[in our FAQ][faq-riot-vs-others] (near the end).
```
Comment (kfessel) [inline README.md:96]:
```suggestion
[in our FAQ][faq-riot] (near the end).
```

File: README.md (hunk 1 above)
Comment (kfessel) [reply README.md]: `#faq-12` seems to work

**Comment by OlegHahm** [comment]
Personally, I would like to get rid of the related projects (except for Ariel). The selection of projects seems quite arbitrary to me and the descriptions seem to be somewhat inconsistent.

If there is consensus on keeping this section with all these OS' we should also define which projects should be mentioned here. Also, I would move this section down. At least after the Getting RIOT section.

**Comment by kfessel** [comment]
> get rid of the related projects (except for Ariel)
My first commit was  as a small friendly note at the end of the intro - i also think it was quiet low key if looked at the README and not just the change in isolation.

> selection of projects seems quite arbitrary 
I think the "related projects" selection is no more random than the FAQ comparison. 
FreeRTOS and Arduino are exchanged for NuttX and RT-Thread

> move down
so right between 
### Getting Riot 
and
### Getting Started 

or between

### Getting Started 
and
### Community

**Comment by OlegHahm** [comment]
> > get rid of the related projects (except for Ariel)
> 
> My first commit was as a small friendly note at the end of the intro - i also think it was quiet low key if looked at the README and not just the change in isolation.

And I liked that commit - except that I still think that the reference to Ariel is misplaced in the intro. The intro describes the key properties and attributes of RIOT - so it felt weird to suddenly have a link to something else (no matter how close or remote we consider the relationship with Ariel OS.
 
> > selection of projects seems quite arbitrary
> 
> I think the "related projects" selection is no more random than the FAQ comparison. FreeRTOS and Arduino are exchanged for NuttX and RT-Thread

Agreed. We need some sort of a definition to what we want to list.

> > move down
> 
> so right between
> ### Getting Riot
> 
> and
> ### Getting Started
> 
> or between
> ### Getting Started
> 
> and
> ### Community
> 
> [kfessel:p-ad-ariel-lowkey](https://github.com/kfessel/RIOT/tree/p-ad-ariel-lowkey) [kfessel:p-ad-ariel-lowkey2](https://github.com/kfessel/RIOT/tree/p-ad-ariel-lowkey2)

Hm. On a second thought I'm not sure. On the one hand I want the link to Ariel OS to be as visible as possible - on the other hand the most natural place for me seems to be rather towards the end of the document.

**Comment by maribu** [comment]
> Personally, I would like to get rid of the related projects (except for Ariel). The selection of 
projects seems quite arbitrary to me and the descriptions seem to be somewhat inconsistent.

Is the issue you have only the selection to be arbitrary and the inconsistency in wording or their description? If so, that seems to be a case for just refining this section.

Or is the issue you have the addition of the related projects section as such, and you would like to see something like "Other Projects from the Same Crowd" instead?

(It would be odd if the section header would be just "Related Projects" or "Other Projects", but only stuff from (parts) of our community would qualify.)

If it is the latter, I'd convert this PR to a draft and revisit if this still makes sense when this discussion has concluded.

**Comment by waehlisch** [comment]
honestly. whatever is about advertising other projects should go to the very end (if such text should be part of the README at all). it appears weird that the RIOT README starts with watering the focus on RIOT. in very former times, a readme file helped the user to quickly get an overview about the software project. the README file is not the web page of the project, even though it feeds the landing page on github.

if you want to educate users that there are other projects that fits better the needs of the user, if you want to help the user to wisely choose their programming language or operating system, the README is not the best place.

**Comment by maribu** [comment]
@waehlisch Please stay on point: This PR is about adding a reference to the FAQ.

**Comment by waehlisch** [comment]
@maribu i tried to stay on point. sorry for misreading the comments. my understanding was different because Oleg wrote "except that I still think that the reference to Ariel is misplaced in the intro"

**Comment by OlegHahm** [comment]
> > Personally, I would like to get rid of the related projects (except for Ariel). The selection of
> > projects seems quite arbitrary to me and the descriptions seem to be somewhat inconsistent.
> 
> Is the issue you have only the selection to be arbitrary and the inconsistency in wording or their description? If so, that seems to be a case for just refining this section.
> 
> Or is the issue you have the addition of the related projects section as such, and you would like to see something like "Other Projects from the Same Crowd" instead?

I'm not sure how helpful it really is to add pointers to other references to the README at all. I tend to agree with @waehlisch here, that something likes this would be better suited for the web page instead. However, I'm probably a bit old-fashioned here (typically rather open the README in a pager application in my terminal emulator then looking at it on Github).

However, in any case if the consensus is that we really want to list third-party projects here, we should have a good definition of the selection criteria first. Why do we list Nuttx but not FreeRTOS? Why RT-Thread but not Frosted or TockOS?

**Comment by maribu** [comment]
Closing in favor of https://github.com/RIOT-OS/RIOT/pull/21078

**Pull request closed without merging**